java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/LambdaExprContext.java

```diff
@@ -65,15 +65,42 @@ public LambdaExprContext(LambdaExpr wrappedNode, TypeSolver typeSolver) {
                         MethodUsage methodUsage = JavaParserFacade.get(typeSolver).solveMethodAsUsage(methodCallExpr);
                         int i = pos(methodCallExpr, wrappedNode);
                         Type lambdaType = methodUsage.getParamTypes().get(i);
-                        Type argType = lambdaType.asReferenceType().typeParametersValues().get(0);
-                        LambdaConstraintType conType;
-                        if (argType.isWildcard()){
-                            conType = LambdaConstraintType.bound(argType.asWildcard().getBoundedType());
-                        } else {
-                            conType = LambdaConstraintType.bound(argType);
+
+                        // Get the functional method in order for us to resolve it's type arguments properly
+                        Optional<MethodUsage> functionalMethodOpt = FunctionalInterfaceLogic.getFunctionalMethod(lambdaType);
+                        if (functionalMethodOpt.isPresent()){
+                            MethodUsage functionalMethod = functionalMethodOpt.get();
+                            InferenceContext inferenceContext = new InferenceContext(MyObjectProvider.INSTANCE);
+
+                            // Resolve each type variable of the lambda, and use this later to infer the type of each
+                            // implicit parameter
+                            inferenceContext.addPair(lambdaType, new ReferenceTypeImpl(lambdaType.asReferenceType().getTypeDeclaration(), typeSolver));
+
+                            // Find the position of this lambda argument
+                            boolean found = false;
+                            int lambdaParamIndex;
+                            for (lambdaParamIndex = 0; lambdaParamIndex < wrappedNode.getParameters().size(); lambdaParamIndex++){
```

Comment:
Tried changing it but it doesn't work because the index gets incremented an extra time which leads to having to do `lambdaParamIndex-1` which is less readable